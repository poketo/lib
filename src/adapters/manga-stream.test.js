import poketo from '..';
import adapter from './manga-stream';

describe('MangaStreamAdapter', () => {
  describe('parseUrl', () => {
    it('returns the components of a url', () => {
      expect(
        adapter.parseUrl('https://readms.net/manga/attack_on_titan'),
      ).toEqual({
        seriesSlug: 'attack_on_titan',
        chapterSlug: null,
      });

      expect(
        adapter.parseUrl('https://readms.net/r/attack_on_titan/103/4949/3'),
      ).toEqual({
        seriesSlug: 'attack_on_titan',
        chapterSlug: '103/4949',
      });
    });

    it('throws on unparseable urls', () => {
      expect(() => {
        adapter.parseUrl('https://readms.net/about');
      }).toThrow(poketo.InvalidUrlError);
    });
  });

  describe('constructUrl', () => {
    it('returns a valid url', () => {
      expect(adapter.constructUrl('attack_on_titan')).toEqual(
        `${adapter._getHost()}/manga/attack_on_titan`,
      );

      expect(adapter.constructUrl('attack_on_titan', '103/4949')).toEqual(
        `${adapter._getHost()}/r/attack_on_titan/103/4949`,
      );
    });
  });
});

describe('MangaStream', () => {
  const server = new AdapterVcrServer(adapter);

  beforeAll(async () => {
    await server.listenAndMock();
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', async () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'manga-stream:shokugeki_no_souma',
      );

      expect(metadata).toMatchSnapshot({
        updatedAt: expect.any(Number),
      });
    });
  });

  describe('getChapter', async () => {
    it('returns a list of pages', async () => {
      const chapter = await poketo.getChapter(
        'manga-stream:a_trail_of_blood:15/4622',
      );

      expect(chapter.pages).toMatchSnapshot();
    });
  });
});
