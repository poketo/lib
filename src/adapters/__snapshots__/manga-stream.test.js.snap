// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangaStreamAdapter getChapter returns a list of pages 1`] = `
Array [
  Object {
    "height": 1200,
    "id": "01.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/01.png",
    "width": 789,
  },
  Object {
    "height": 800,
    "id": "01a.jpg",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/01a.jpg",
    "width": 800,
  },
  Object {
    "height": 1200,
    "id": "02.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/02.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "03.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/03.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "04.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/04.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "05.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/05.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "06.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/06.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "07.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/07.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "08.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/08.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "09.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/09.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "10.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/10.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "11.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/11.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "12.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/12.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "13.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/13.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "14-15.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/14-15.png",
    "width": 1581,
  },
  Object {
    "height": 1200,
    "id": "16.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/16.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "17.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/17.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "18.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/18.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "19.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/19.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "20.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/20.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "21.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/21.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "22.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/22.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "23.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/23.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "24.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/24.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "25.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/25.png",
    "width": 789,
  },
  Object {
    "height": 1200,
    "id": "261.png",
    "url": "https://img.mangastream.com/cdn/manga/149/4622/261.png",
    "width": 789,
  },
]
`;

exports[`MangaStreamAdapter getSeries returns a metadata object 1`] = `
Object {
  "slug": "shokugeki_no_souma",
  "title": "Shokugeki no Souma",
  "url": "https://readms.net/manga/shokugeki_no_souma",
}
`;

exports[`MangaStreamAdapter getSeries returns a metadata object 2`] = `
Array [
  Object {
    "chapterNumber": "264",
    "createdAt": 1527206400,
    "slug": "264/5101",
    "title": "From: Tadokoro Megumi",
    "url": "https://readms.net/r/shokugeki_no_souma/264/5101",
  },
]
`;
